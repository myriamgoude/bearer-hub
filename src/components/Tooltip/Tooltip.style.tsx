import css from '@emotion/css'

export default {
  root: css`
    vertical-align: middle;
  `
}
