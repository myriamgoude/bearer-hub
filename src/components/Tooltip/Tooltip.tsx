import React from 'react'
import RCTooltip from 'rc-tooltip'

import 'rc-tooltip/assets/bootstrap.css'
import styles from './Tooltip.style'

import tooltipImg from '../../images/shared/icon-question.svg'

interface ITooltipProps {
  content: string
  placement: string
  trigger: any
}

const Tooltip = (props: ITooltipProps) => (
  <RCTooltip
    id={props.content}
    overlay={props.content}
    placement={props.placement}
    trigger={props.trigger}
    css={styles.root}
  >
    <span aria-describedby={props.content}>
      <img src={tooltipImg} />
    </span>
  </RCTooltip>
)

Tooltip.defaultProps = {
  placement: 'right',
  trigger: ['focus', 'hover']
}

export default Tooltip
