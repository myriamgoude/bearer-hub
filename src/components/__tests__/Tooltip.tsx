import * as React from 'react'
import renderer from 'react-test-renderer'
import Tooltip from '../Tooltip/Tooltip'

describe('Tooltip', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Tooltip placement="right" trigger={['alert']} content="Did you know this is a tooltip?" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
