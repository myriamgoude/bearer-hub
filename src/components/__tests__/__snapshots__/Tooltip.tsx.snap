// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly 1`] = `
<span
  aria-describedby="Did you know this is a tooltip?"
>
  <img
    src="test-file-stub"
  />
</span>
`;
